make ActivityRoute Parcelable

gabrielittner opened the pull request:
Currently it is not `Parcelable` because putting it into the `Intent` would crash any third party app launched by it. So consumers can make it Parcelable for internal use and then we will add it to the extras. The issue is that this makes deep links a lot more complex because we can't pass through the route and then handle it. With this `ActivityRoute` becomes `Parcelable` and the default implementation of `fillInIntent` will add it to the extras. Other places like the activity navigator don't need to deal with it anymore. When implementing route for external destinations users can override `fillInIntent` to prevent the route from being added and therefor avoid the crash.

kboyarshinov commented on navigator/runtime/src/main/java/com/freeletics/mad/navigator/NavRoute.kt:
```diff
@@ -28,15 +28,15 @@ public interface NavRoot : BaseRoute
- * When the implementing class is [Parcelable], the instance of route will be put into the
- * navigation arguments and is then available to the target screens. Do not do this in case this
- * route leads to a different app.
+ * By default the instance of this class will be added to th resulting `Intent` and can be accessed
```
```suggestion
 * By default the instance of this class will be added to the resulting `Intent` and can be accessed
```

kboyarshinov reviewed (approved):
LGTM :+1: 

what do you think of adding something like `ExternalActivityRoute` to facilitate corrent external activity use case?

```kotlin
public abstract class ExternalActivityRoute : ActivityRoute {
    final override fun fillInIntent(): Intent {
        return externalIntent()
    }
    
    public abstract fun externalIntent(): Intent
}
```

gabrielittner commented:
> what do you think of adding something like ExternalActivityRoute to facilitate corrent external activity use case?

Added it but in a slightly different way. I've made `ActivityRoute` sealed and then also added `InternalActivityRoute`. This one has a final override of fillInIntent which means it is guaranteed that these have the route extra. `ExternalActivityRoute` then has the same default implementation with an empty intent that we had on the base interface before.

gabrielittner commented:
snapshot publishing failed because of a 502 from sonatype
